frame/staking-rewards/src/lib.rs

```diff
@@ -453,6 +481,20 @@ pub mod pallet {
 														)?,
 													)
 													.map_err(|_| ArithmeticError::Overflow)?;
+												if let Some(amount) =
+													PendingAmountExtensions::<T>::take(&nft_id)
+												{
+													let time_lock = honest_locked_stake_increase(
+														nft.early_unstake_penalty.value,
+														nft.stake.into(),
+														amount.into(),
+														nft.lock_duration.into(),
+														 (now - nft.lock_date).into())
+														 .ok_or(Error::<T>::CannotIncreaseStakedAmountBecauseOfLimitedArithmetic)?;
+													nft.lock_date =
+														nft.lock_date.safe_add(&time_lock)?;
+													nft.stake = nft.stake.safe_add(&amount)?;
+												}
```

Comment:
I think you need to do that on the registering phase as you are increasing the share of the user while rewarding the other users, i.e. you dillute the others

Reply:
Also not sure why we do a complex computation, we should just refresh the lock, i.e. `lock_date = date(call(extend_stake))` ?

Reply:
> I think you need to do that on the registering phase as you are increasing the share of the user while rewarding the other users, i.e. you dillute the others

huh, exactly what I am fixing. i do not increase total stake. I will increase pending shared stake merged on registering. right now it does not dillutes for sure. but problem is that it does not increase. if fixing increase.

Reply:
> Also not sure why we do a complex computation, we should just refresh the lock, i.e. lock_date = date(call(extend_stake)) ?

if you extend statke without using time of lock as value you incentivize creating new nft because old one captures some value lost during extensions. 

i.e. setting date of nft the date is extended will not be used by bots. as it will loose value.

Reply:
so if formula not complex, than it is same as not having extension at all technically as people better create more accounts and fnfts than so the value is rolled.